torchtitan/datasets/experimental/llama3/meta/c4_llama3_counts.csv

```diff
@@ -0,0 +1,2 @@
+dataset/filename,documents,tokens
+/c4_mini/c4_mini.arrow,45000,20505558
```

Comment:
1. The expected format is for each document to occupy one row in the pyarrow file. Because docs are different lengths, there's not really any consistent notion of columns anymore (or viewed another way, the doc rows all occupy the single "tokens" column of the shard file)
2. Right now we pretokenize to enable chunking - currently, we can yield portions of a document at a time, without ever having to load the whole doc into memory at once, which is helpful in the case of extremely long docs. This gets more difficult for raw text since we have to split on tokens rather than on characters, but we don't know the token boundaries until after tokenization. Support for dynamic tokenization is on the todo list though!